src/S3/MultipartCopy.php

```diff
@@ -127,7 +127,8 @@ private function createPart($partNumber, $partsCount)
             $data[$k] = $v;
         }
 
-        $data['CopySource'] = $this->source;
+        list($bucket, $key) = explode('/', ltrim($this->source, '/'), 2);
```

Comment:
Since this is adding new source handling logic, should there be tests to make sure that different inputs for source behave as expected (i.e. multiple slashes, etc)?

Reply:
You're right, and just to be safe I switched the code up a bit and added some more tests